src/MinterGateway.sol

```diff
@@ -1051,20 +1051,21 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
         uint256[] calldata timestamps_,
         bytes[] calldata signatures_
     ) internal view returns (uint40 minTimestamp_) {
-        uint256 threshold_ = updateCollateralValidatorThreshold();
+        if (signatures_.length != updateCollateralValidatorThreshold()) {
+            revert InvalidSignatureCount(signatures_.length, updateCollateralValidatorThreshold());
+        }
 
         minTimestamp_ = uint40(block.timestamp);
 
-        // Stop processing if there are no more signatures or `threshold_` is reached.
-        for (uint256 index_; index_ < signatures_.length && threshold_ > 0; ++index_) {
+        for (uint256 index_; index_ < signatures_.length; ++index_) {
             unchecked {
                 // Check that validator address is unique and not accounted for
                 // NOTE: We revert here because this failure is entirely within the minter's control.
                 if (index_ > 0 && validators_[index_] <= validators_[index_ - 1]) revert InvalidSignatureOrder();
             }
 
             // Check that validator is approved by TTG.
-            if (!isValidatorApprovedByTTG(validators_[index_])) continue;
+            if (!isValidatorApprovedByTTG(validators_[index_])) revert NotApprovedValidator(validators_[index_]);
```

Comment:
```suggestion
            _revertIfNotApprovedValidator(validators_[index_]);
```

Reply:
Fixed in this commit: https://github.com/MZero-Labs/protocol/pull/165/commits/2671e1ed4b0491b1c3c102e73b86f4429b6b47f3